helpers/Physics.h

```diff
@@ -0,0 +1,321 @@
+#ifndef _HELPERS_PHYSICS_H_
+#define _HELPERS_PHYSICS_H_
+
+#ifndef _Z64HDR_HELPER_PREFIX_
+#define _Z64HDR_HELPER_PREFIX_
+#endif
+
+#include "include/z64.h"
+#include "Matrix.h"
+
+typedef void(*PhysicCallback)(s32 limbIndex, void*, void*);
+
+typedef struct {
+	Vec3f pos;
+	Vec3f rot;
+	Vec3f vel;
+} PhysicsJoint;
+
+typedef struct {
+	s32 numLimbs;
+	f32 gravity;
+	f32 floorY;  // world.pos.y, won't go through this
+	f32 maxVel;  // Clamps velocity value
+	f32 velStep; // Values below 1.0f will give it spring like motion
+	f32 velMult; // Control the power of velocity
+} PhysicsInfo;
+
+typedef struct {
+	Vec3f pos;
+	Vec3s rot;
+	MtxF* mtxF;
+} PhysicsHead;
+
+typedef struct {
+	u32   dlist;
+	Vec3f scale;  // Gfx scale
+	u8    segID;  // For matrix
+	u8    noDraw; /* If there needs to be calculations
+	                 to get in position before drawing */
+} PhysicsGfx;
+
+typedef struct {
+	s32    num; // amount of spheres
+	Vec3f* centers;
+	f32    radius;
+} PhysicsSpheres; // "collision" spheres, pushes limbs away
+
+typedef struct {
+	u8    lockRoot;    // Prevent physics rotating root limb
+	Vec2f rotStepCalc; // DEG, limits rot to next limb in main calc
+	Vec2f rotStepDraw; // DEG, limits rot on draw, smoothens output
+} PhysicsConstraint;
+
+typedef struct {
+	s32   num;  // How many limbs will be smoothed with push
+	Vec3f push; // direction Z, pushes based on rot[0]
+	f32   mult; // How much the pushing fill affect
+	Vec3f rot;  // DEG, rigids towards, relative rot
+} PhysicsRigidity;
+
+typedef struct {
+	PhysicsInfo       info;
+	PhysicsHead       head;
+	PhysicsGfx        gfx;
+	PhysicsSpheres    spheres;
+	PhysicsConstraint constraint;
+	PhysicsRigidity   rigidity;
+	f32 limbsLength[];
+} PhysicsStrandInit;
+
+typedef struct {
+	PhysicsInfo       info;
+	PhysicsHead       head;
+	PhysicsGfx        gfx;
+	PhysicsSpheres    spheres;
+	PhysicsConstraint constraint;
+	PhysicsRigidity   rigidity;
+	f32* limbsLength;
+} PhysicsStrand;
+
+_Z64HDR_HELPER_PREFIX_
+void Physics_GetHeadProperties(PhysicsStrand* strand, Vec3f* mult, s32 flag) {
+	static MtxF mtxF;
+	Vec3f zero = { 0 };
+	
+	Matrix_Get(&mtxF);
+	strand->head.mtxF = &mtxF;
+	func_800D20CC(&mtxF, &strand->head.rot, flag);
```

Comment:
```suggestion
	func_800D20CC(&mtxF, &strand->head.rot, 0);
```
`flag` is always used as 0 in the game
and it usually can be 0 (I documented that function here https://github.com/Dragorn421/oot/blob/goroiwa_doc/src/code/sys_matrix.c#L742 )